src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.ts

```diff
@@ -0,0 +1,201 @@
+import { ColumnData, SortableTableCellData } from '../../components/sortable-table/sortable-table.component';
+import { ErrorMessageOutput } from '../../error-message-output';
+
+/**
+ * Model for searching of logs
+ */
+interface SearchLogsFormModel {
+  logsDateFrom: DateFormat;
+  logsDateTo: DateFormat;
+  logsTimeFrom: TimeFormat;
+  logsTimeTo: TimeFormat;
+  courseId: string;
+  studentName: string;
+  isSearchDisabled: boolean;
+}
+
+/**
+ * Model for displaying of feedback session logs
+ */
+interface FeedbackSessionLogModel {
+  feedbackSessionName: string;
+  logColumnsData: ColumnData[];
+  logRowsData: SortableTableCellData[][];
+  isTabExpanded: boolean;
+}
+
+/**
+ * Component for instructor logs
+ */
+@Component({
+  selector: 'tm-instructor-audit-logs-page',
+  templateUrl: './instructor-audit-logs-page.component.html',
+  providers: [{ provide: NgbDateParserFormatter, useClass: SessionEditFormDatePickerFormatter }],
+  styleUrls: ['./instructor-audit-logs-page.component.scss'],
+})
+export class InstructorAuditLogsPageComponent implements OnInit {
+  LOG_RETENTION_DAYS: number = 30;
+
+  // enum
+  SortBy: typeof SortBy = SortBy;
+
+  formModel: SearchLogsFormModel = {
+    logsDateFrom: { year: 0, month: 0, day: 0 },
+    logsTimeFrom: { hour: 0, minute: 0 },
+    logsDateTo: { year: 0, month: 0, day: 0 },
+    logsTimeTo: { hour: 0, minute: 0 },
+    courseId: '',
+    studentName: '',
+    isSearchDisabled: true,
+  };
+  dateToday: DateFormat = { year: 0, month: 0, day: 0 };
+  earliestSearchDate: DateFormat = { year: 0, month: 0, day: 0 };
+  courses: Course[] = [];
+  courseToStudents: Record<string, Student[]> = {};
+  searchResults: FeedbackSessionLogModel[] = [];
+  isLoading: boolean = true;
+  isSearching: boolean = false;
+
+  constructor(private courseService: CourseService,
+              private studentService: StudentService,
+              private logsService: LogService,
+              private timezoneService: TimezoneService,
+              private statusMessageService: StatusMessageService) { }
+
+  ngOnInit(): void {
+    const today: Date = new Date();
+    this.dateToday.year = today.getFullYear();
+    this.dateToday.month = today.getMonth() + 1;
+    this.dateToday.day = today.getDate();
+
+    const earliestSearchDate: Date = new Date(Date.now() - this.LOG_RETENTION_DAYS * 24 * 60 * 60 * 1000);
+    this.earliestSearchDate.year = earliestSearchDate.getFullYear();
+    this.earliestSearchDate.month = earliestSearchDate.getMonth() + 1;
+    this.earliestSearchDate.day = earliestSearchDate.getDate();
+
+    this.formModel.logsDateFrom = { ...this.dateToday };
+    this.formModel.logsDateTo = { ...this.dateToday };
+    this.formModel.logsTimeFrom = { hour: today.getHours(), minute: 0 };
+    this.formModel.logsTimeTo = { hour: 23, minute: 59 };
+    this.loadData();
+  }
+
+  /**
+   * Search for logs of student activity
+   */
+  search(): void {
+    this.isSearching = true;
```

Comment:
Disable search button here? 

Also, the current method of disabling the search button is very manual. Maybe something like [disabled]="!formModel.courseId || isSearching || isLoading"?